sdk/core/core-http/lib/credentials/tokenCredential.ts

```diff
@@ -11,11 +11,37 @@ export interface TokenCredential {
    * Gets the token provided by this credential.
    *
    * @param scopes The list of scopes for which the token will have access.
-   * @param requestOptions The RequestOptionsBase used to configure any requests
-   *                       this TokenCredential implementation might make.
+   * @param getTokenOptions The options used to configure any requests
+   *                        this TokenCredential implementation might make.
    */
   getToken(
     scopes: string | string[],
-    requestOptions?: RequestOptionsBase
-  ): Promise<string | null>;
+    getTokenOptions?: GetTokenOptions
+  ): Promise<AccessToken | null>;
+}
+
+/**
+ * Defines options for TokenCredential.getToken.
+ */
+export interface GetTokenOptions {
+  /**
+   * An AbortSignalLike implementation that can be used to cancel
+   * the token request.
+   */
+  abortSignal?: AbortSignalLike;
+}
+
+/**
+ * Represents an access token with an expiration time.
+ */
+export interface AccessToken {
+  /**
+   * The access token.
+   */
+  token: string;
+
+  /**
+   * The access token's expiration date and time.
+   */
+  expiresOn: Date;
```

Comment:
Is this UTC? I think there is a guideline somewhere saying that date arguments should have the "utc" suffix

Reply:
JavaScript stores its Date internally as a UNIX timestamp in UTC but it appears to be in the local timezone when you call the `getDate()` etc methods (you have to use `getUTCDate()` explicitly to get UTC).  Since the date appears to be in the local timezone I think it'd be more confusing to have a `Utc` suffix.

Is there a `Local` suffix or something like it when the time isn't in UTC?  I could use that instead

Reply:
I'll defer this to @bterlson :)

Reply:
Spoke with @bterlson and he thinks that since `Date` is represented in the local time zone that we shouldn't use a suffix in this case.  If the team comes up with a good guideline for this I'll come back and change this before Preview